src/Approximations/iterative_refinement.jl

```diff
@@ -56,19 +56,21 @@ Type that represents the polygonal approximation of a convex set.
 - `constraints`  -- vector of linear constraints that are already finalized
                     (i.e., they satisfy the given error bound)
 """
-struct PolygonalOverapproximation{N<:Real}
-    S::LazySet{N}
-    approx_stack::Vector{LocalApproximation{N}}
-    constraints::Vector{LinearConstraint{N, Vector{N}}}
+struct PolygonalOverapproximation{N<:Real, SN<:LazySet{N}, VN<:AbstractVector{N}}
+    S::SN
+    approx_stack::Vector{LocalApproximation{N, VN}}
+    constraints::Vector{LinearConstraint{N, VN}}
+end
 
-    function PolygonalOverapproximation(S::LazySet{N}) where {N<:Real}
-        return new{N}(S, Vector{LocalApproximation{N}}(), Vector{LinearConstraint{N,Vector{N}}}())
-    end
+function PolygonalOverapproximation(S::SN) where {N<:Real, SN<:LazySet{N}}
+    empty_local_approx = Vector{LocalApproximation{N, Vector{N}}}()
+    empty_constraints = Vector{LinearConstraint{N,Vector{N}}}()
+    return PolygonalOverapproximation(S, empty_local_approx, empty_constraints)
```

Comment:
The parameter `SN` is not used. If that is correct, then I do not understand why you added it.

Reply:
Sorry, i didn't understand your comment.

Reply:
Oh, do you mean that you'd like to see `PolygonalOverapproximation(S::LazySet{N}) where {N<:Real}` as before instead of `PolygonalOverapproximation(S::SN) where {N<:Real, SN<:LazySet{N}}` ? I added it as a reflex, and these forms are equivalent as far as i know.

Reply:
> I added it as a reflex

Okay. I mentioned it because often a change with no effect is suspicious and indicates that something was forgotten.

> these forms are equivalent as far as i know

Yes, they are equivalent. I find the new version slightly harder to read (because I have to look to the right in order to find out the dispatch) but we can keep it.

Reply:
Ok. If you make the suggestion i would approve it, otherwise, either version is fine to me.